clients/spark/core/src/main/scala/io/treeverse/clients/ApiClient.scala

```diff
@@ -27,8 +32,22 @@ private object ApiClient {
               uri.getQuery,
               uri.getFragment
              )
-    else
+    } else if (storageType == StorageTypeAzure) {
+
+      /** regex to catch url of type: https://StorageAccountName.blob.core.windows.net/Container[/BlobName] and convert it to abfs url */
+      val regex = "https?://([^.]+)(\\.[^/]+)(?:/([^/]+)(.+))?"
+      val pattern = Pattern.compile(regex)
+      val matcher = pattern.matcher(uri.toString())
+      if (matcher.find()) {
+        return new URI(
+          s"abfs://${matcher.group(3)}@${matcher.group(1)}.dfs.core.windows.net${matcher.group(4)}"
+        )
+      }
```

Comment:
This approach assumes that Azure URLs have a particular form.  But a quick search seemed to reveal [the host can be modified](https://docs.microsoft.com/en-us/azure/storage/blobs/storage-custom-domain-name?tabs=azure-portal).  I do not thing it will work in that case.

Reply:
@lynnro314 to verify if lakeFS supports this URL modification, we should try to create a lakeFS repository with a storage namespace and see if lakeFS support that at all. If this doesn't work I think that we are good to go because GC only deletes objects that lakeFS wrote to a repo storage namespace. @arielshaqed WDYT?

(The suspect that may fail this this https://github.com/treeverse/lakeFS/blob/2358e8dd7f72af67844413f7177b7e3c94ee1c71/pkg/block/azure/adapter.go#L175 line of code we've looked at together)

Reply:
Yes, but please bear in mind:

1. If _we_ cannot configure it to work, that does not mean a greater Azure expert might not be able to configure it to work :-)
1. It might be fixed in lakeFS someday, we might not know (or I will know it was fixed but forget this PR), and then GC will stop working.

So I agree that if it is hard to do correctly right now we do not have to do it, but if it is easy to do then I would rather we tried.